src/app/metadata/features/dataset-expansion-panel/dataset-expansion-panel.component.spec.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Test the dataset expansion panel
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { DatasetExpansionPanelComponent } from './dataset-expansion-panel.component';
+
+describe('DatasetExpansionPanelComponent', () => {
+  let component: DatasetExpansionPanelComponent;
+  let fixture: ComponentFixture<DatasetExpansionPanelComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [DatasetExpansionPanelComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(DatasetExpansionPanelComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```

Comment:
This and the other tests currently fail because you don't provide the HttpClient in the test bed which is needed by the component.

The component should depend on a custom MetadataService instead. In the test, you can then provide a MockMetadataService which has only the method that is required by that individual component, returning some fake data (maybe import and reuse the mock data). You can then test that the fake data appears on the screen (maybe just as a spot check for one data field). Keep it very simple.

Reply:
See the site-header.component.spec.ts for an example.

Reply:
There are still issues that I have no clue why they fail. It seems zoneJS doesn't like the new signal calling syntax?

Reply:
No, zone.js is not a problem since we're not using it.

The error messages from the tests should give you a clue. As far as I see they are failing because the mock service does not provided the necessary methods.

Reply:
Sorry I meant Jest. I assumed that the methods would simply be gotten from the service itself, since we provide it.

Reply:
Done 👍